JS-Algorithms/challenges.ts

```diff
@@ -40,8 +63,14 @@ Invoking "circularArray(2)" should return "["Island", "Japan", "Israel", "German
 
 const COUNTRY_NAMES = ["Germany", "Norway", "Island", "Japan", "Israel"];
 
-const circularArray = (index) => {
+const circularArray = (index: number): string[] => {
   // YOUR CODE HERE...
+  const len = COUNTRY_NAMES.length;
+  const newIndex = ((index % len) + len) % len;
+  let newArray = COUNTRY_NAMES.slice(newIndex, len);
+  newArray = newArray.concat(COUNTRY_NAMES.slice(0, newIndex));
+
+  return newArray;
```

Comment:
Seems you can probably just keep chaining methods here and avoid having to come up with new variable names. Try to avoid names like "new" when naming variables.

`return COUNTRY_NAMES.slice(newIndex, len).concat(COUNTRY_NAMES.slice(0, newIndex))`